### PR TITLE
add custom params to deploy demo

### PR DESCRIPTION
**Pull request by jmhauck**
Allows us to pass custom params like wkid or the upcoming feature service id.

### CODE REVIEW

File: demos/src/deploy-solution-main.ts [hunk 1]
```diff
@@ -131,15 +138,19 @@ export function deployAndDisplaySolution(
...
   // Deploy a solution described by the supplied id
   const options: common.IDeploySolutionOptions = {
     jobId: templateSolutionId,
     progressCallback: progressCallback,
     consoleProgress: true,
     storageAuthentication: srcAuthentication,
-    enableItemReuse
+    enableItemReuse,
+    templateDictionary: {
+      params: JSON.parse(customParams)
```
Comment (MikeTschudi) [inline demos/src/deploy-solution-main.ts:152]: If a solution template is supplied and the Custom Params input box is left alone, "Deploying..." appears on the screen, the app hangs, and the debugger shows "Uncaught SyntaxError: Unexpected end of JSON input".

File: demos/src/deploy-solution-main.ts [hunk 2]
```diff
@@ -107,15 +110,19 @@ export function deploySolution(
...
   // Deploy a solution described by the supplied id
   const options: common.IDeploySolutionOptions = {
     jobId,
     progressCallback: progressCallback,
     consoleProgress: true,
     storageAuthentication: srcAuthentication,
-    enableItemReuse
+    enableItemReuse,
+    templateDictionary: {
+      params: JSON.parse(customParams)
```
Comment (MikeTschudi) [inline demos/src/deploy-solution-main.ts:124]: If a folder of solution templates is supplied and the Custom Params input box is left alone, "Deploying..." appears on the screen and then is replaced by "{}".

**Review by MikeTschudi — CHANGES_REQUESTED** [review]
The Custom Params box seems as if it should be optional if one doesn't have custom parameters, but this doesn't work: "{}" has to be supplied as a minimum. Is this a requirement? Can we change the web page to something like
```js
const customParams = document.getElementById("customParams").value || "{}";
```
so that the custom parameters have a default value?

**Comment by jmhauck** [comment]
Thanks @MikeTschudi

**Review by MikeTschudi — APPROVED** [review]

**Pull request merged**